External scheduler: don't log or send credentials

PhilippMatthes opened the pull request:
The external scheduler call logged two types of credentials:

1. DB and mqtt connection secrets nested in the cell mapping
2. The user's auth token from the request context

This change removes the credentials from logging and also
avoids sending them to the external scheduler.

joker-at-work commented on nova/scheduler/external.py:
```diff
@@ -63,11 +63,22 @@ def call_external_scheduler_api(context, weighed_hosts, weights, spec_obj):
+    if "requested_destination" in spec_data:
+        if "cell" in (req_destination := spec_data["requested_destination"]):
+            del req_destination["cell"]["database_connection"]
+            del req_destination["cell"]["mqtt_credentials"]
```
`transport_url` should be the name

PhilippMatthes committed:
External scheduler: don't log credentials

The external scheduler call logged two types of credentials:
1. DB and mqtt connection secrets nested in the cell mapping
2. The user's auth token from the request context
This change removes the credentials from logging and also
avoids sending them to the external scheduler.
